### PR TITLE
[WIP] SpriteFrames: added option for frame size when importing from a sheet.

### PR DESCRIPTION
**Pull request by dankan1890**
Godot 2020-01-06 053504

### CODE REVIEW

**Comment by groud** [comment]
The idea makes sense, but I don't think the implementation is great. I would:
- Remove the selection mode,
- Simply update the other corresponding label when you change a value.
or 
- Keep the selection mode,
- Hide/show the other input fields depending on the selected mode.

**Comment by dankan1890** [comment]
@groud 
I too had thought about the second option (you have to excuse me but I didn't understand the first one), but then I opted to leave everything so that the user can work on the various values regardless of the selector and switch from one mode to another already having an idea of the result.

**Comment by groud** [comment]
> I opted to leave everything so that the user can work on the various values regardless of the selector and switch from one mode to another already having an idea of the result.

That's not a really good idea UX-wise, changing a value should have an impact on the reasult, otherwise it's going to be very confusing for users.

> you have to excuse me but I didn't understand the first one

When you type a number in the "horizontal" input field, the "width" field is automatically updated with the corresponding computed value, and vice versa. Same for the "vertical" and "height" fields.

File: editor/plugins/sprite_frames_editor_plugin.cpp [hunk 1]
```diff
@@ -200,20 +228,52 @@ void SpriteFramesEditor::_sheet_spin_changed(double) {
...
+			split_sheet_pixel_vb->hide();
+			break;
+		case MODE_PIXEL:
+			split_sheet_grid_vb->hide();
+			split_sheet_pixel_vb->show();
+			break;
+		default:
+			// It shouldn't happen
+			break;
+	}
```
Comment (groud) [inline editor/plugins/sprite_frames_editor_plugin.cpp]: Maybe you could simplify this part of the code with something like :
```
split_sheet_grid_vb->set_visible(mode==MODE_GRID)
split_sheet_pixel_vb->set_visible(mode==MODE_PIXEL)
```

Comment (dankan1890) [reply editor/plugins/sprite_frames_editor_plugin.cpp]: Ok, thanks :)

**Comment by dankan1890** [comment]
Unfortunately I noticed that there are general inaccuracies (in the selection, in generating the grid, etc).
Sorry.
For now I close to work better, I will reopen it when stable.

Ps: thanks to @groud  for the review and support :)

**Pull request closed without merging**